pkg/cases/handler.go

```diff
@@ -8,14 +8,125 @@ import (
+
 // @Summary Get case by case id
 // @Description
+// @Tags Case
 // @Produce json
 // @Param  id path uint true "case ID"
 // @Success 200 {object} api.ReturnedData{data=Case}
 // @Router /case/{id} [GET]
 func (h *CaseHandler) GetCaseByCaseID(c echo.Context) error {
 	// what is in here does not matter for now, just filling the swaggo comments and name the function
 	c.Logger().Debug("hello world")
-	return c.JSON(200, api.Return("pong", nil))
+	return c.JSON(200, api.Return("GetCaseByCaseID", nil))
+}
+
+// @Summary Get prevoius case by current case ID
+// @Description
+// @Tags Case
+// @Produce json
+// @Param  id path uint true "current case ID"
+// @Success 200 {object} api.ReturnedData{data=Case}
+// @Router /case/{cid} [GET]
+func (h *CaseHandler) GetPreviousCaseByID(c echo.Context) error {
+	// ...
+	c.Logger().Debug("hello world")
+	return c.JSON(200, api.Return("GetPreviousCaseByID", nil))
+}
+
+// @Summary Get lastest case by department name
+// @Description
+// @Tags Case
+// @Produce json
+// @Param  department path uint true "department name"
+// @Success 200 {object} api.ReturnedData{data=Case}
+// @Router /case/department [GET]
+func (h *CaseHandler) GetPreviousCaseByDepartment(c echo.Context) error {
+	// ...
+	c.Logger().Debug("hello world")
+	return c.JSON(200, api.Return("GetPreviousCaseByDepartment", nil))
+}
+
+// @Summary Update complaint
+// @Description
+// @Tags Case
+// @Produce json
+// @Param id formData uint true "case ID"
+// @Param details formData string true "complaint details"
+// @Success 200 {string} string "{"msg":"modification success"}"
+// @Router /case/complaint/{id} [PATCH]
+func (h *CaseHandler) UpdateComplaint(c echo.Context) error {
+	// ...
+	c.Logger().Debug("hello world")
+	return c.JSON(200, "modification success")
+}
+
+// @Summary Update diagnosis
+// @Description
+// @Tags Case
+// @Produce json
+// @Param id formData uint true "case ID"
+// @Param details formData string true "diagnosis details"
+// @Success 200 {string} string "{"msg":"modification success"}"
+// @Router /case/diagnosis/{id} [PATCH]
+func (h *CaseHandler) UpdateDiagnosis(c echo.Context) error {
+	// ...
+	c.Logger().Debug("hello world")
+	return c.JSON(200, "modification success")
+}
+
+// @Summary Update Past History
+// @Description
+// @Tags Case
+// @Produce json
+// @Param id formData uint true "case ID"
+// @Param details formData string true "Past History details"
+// @Success 200 {string} string "{"msg":"modification success"}"
+// @Router /case/PastHistory/{id} [PATCH]
+func (h *CaseHandler) UpdatePastHistory(c echo.Context) error {
+	// ...
+	c.Logger().Debug("hello world")
+	return c.JSON(200, "modification success")
+}
+
+// @Summary New a prescrition
+// @Description
+// @Tags Prescription
+// @Produce json
+// @Param id formData uint true "case ID"
+// @Param details formData string true "prescription details"
+// @Success 200 {string} string "{"msg":"establishment success"}"
+// @Router /prescription [POST]
```

Comment:
It would be better to place it under some `/case/{caseid}/prescription [POST]`. After all a prescription has to belong to a case.